src/SampleDriver.cpp

```diff
@@ -0,0 +1,273 @@
+    result.mutable_translation_mm()->add_data(0.0);
+
+    // Identity quaternion
+    result.mutable_rotation()->add_data(0.0);
+    result.mutable_rotation()->add_data(0.0);
+    result.mutable_rotation()->add_data(0.0);
+    result.mutable_rotation()->add_data(1.0);
+
+    result.set_config("");
+
+    return result;
+}
+
+static ral::Twist zeroTwist()
+{
+    ral::Twist result;
+
+    // No translation velocity
+    result.mutable_linear_mm_per_s()->add_data(0.0);
+    result.mutable_linear_mm_per_s()->add_data(0.0);
+    result.mutable_linear_mm_per_s()->add_data(0.0);
+
+    // No rotation velocity
+    result.mutable_angular_rad_per_s()->add_data(0.0);
+    result.mutable_angular_rad_per_s()->add_data(0.0);
+    result.mutable_angular_rad_per_s()->add_data(0.0);
+
+    return result;
+}
+
+static ral::JointData zeroJointData()
+{
+    ral::JointData result;
+
+    // No translation
+    result.add_data(0.0);
+    result.add_data(0.0);
+    result.add_data(0.0);
+    result.add_data(0.0);
+    result.add_data(0.0);
+    result.add_data(0.0);
+
+    return result;
+}
+
+namespace ready {
+SampleDriver::SampleDriver(const device::Configuration &configuration, QObject *parent) : Driver(configuration, parent)
+{}
+
+void SampleDriver::start(const ral::RequiredSignals & /*unused */)
+{
+    static constexpr int NUM_JOINTS = 6;
+    static constexpr auto DIGITAL_INPUT_KEY = "DIN";
+    static constexpr auto DIGITAL_OUTPUT_KEY = "DOUT";
+    static constexpr int NUM_DIGITAL_INPUT = 6;
+    static constexpr int NUM_DIGITAL_OUTPUT = 6;
+
+    const auto now = forge::common::utcTime();
+
+    ral::CartesianState cartesian_state;
+    *cartesian_state.mutable_timestamp() = now;
+    *cartesian_state.mutable_pose() = identityPose();
+    *cartesian_state.mutable_velocity() = zeroTwist();
+
+    ral::JointState joint_state;
+    *joint_state.mutable_timestamp() = now;
+    *joint_state.mutable_position() = zeroJointData();
+    *joint_state.mutable_velocity() = zeroJointData();
+    *joint_state.mutable_effort() = zeroJointData();
+    emit armStateChanged(cartesian_state, joint_state);
+
+    ral::DriverState driver_state;
+    *driver_state.mutable_timestamp() = now;
+    driver_state.set_operational_mode(ral::State::MANUAL_REDUCED);
+    driver_state.set_control_mode(ral::State::REMOTE);
+    *driver_state.mutable_guide_mode() = {};
+    driver_state.set_motion_possible(true);
+    driver_state.set_in_motion(false);
+    driver_state.set_enabling_switch_held(true);
+    driver_state.set_can_run(true);
+    driver_state.set_fence_closed(true);
+    driver_state.mutable_speed()->set_speed_factor(1.0);
+    driver_state.set_safety_override(ral::State::NONE);
+    emit controllerStateChanged(driver_state);
+
+    ral::DriverConfiguration driver_config;
+    driver_config.set_controller_model("RCU");
+    driver_config.set_robot_model("READY-5.0");
+
+    // io_configurationuration
+    auto &io_configuration = *driver_config.mutable_io_configuration();
```

Comment:
Do we really need to have them create an I/O signal to start? I guess the only downside is that you will get an I/O mismatch warning down the road.

Reply:
No, I don't think it's required to start up. Copied from comau. I'll remove it.

Reply:
Addressed.